datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -903,9 +992,14 @@ impl GroupedHashAggregateStream {
             && matches!(self.mode, AggregateMode::Partial)
             && self.update_memory_reservation().is_err()
         {
-            let n = self.group_values.len() / self.batch_size * self.batch_size;
-            let batch = self.emit(EmitTo::First(n), false)?;
-            self.exec_state = ExecutionState::ProducingOutput(batch);
+            if !self.enable_blocked_group_states {
```

Comment:
Here is an example of spilling logic mixes with this PR's implementation (see review summary) 🤔

Reply:
I have removed codes added for trying to support this optimization spilling.
We should fix the exist problems in spilling first, and consider a efficient way about supporting.